### DISCUSSION TITLE
Bite sensor on the Haxophone?

### DISCUSSION THREAD

**Discussion by cdsoftw** [cardonabits/haxo-hw#52] [Ideas]
I'm a bit of a hw/electronics novice, but this would be my dream instrument if it supported lip bends using a bite sensor, similar to the (discontinued) Yamaha WX line of EWIs or Roland Aerophone. How feasible would this be to implement (preferably using something off-the-shelf), either by myself or the original creators?

**Comment by jcard0na** [comment]
Hi Cole,

I am not planning to add a bite sensor at this time, but I'd be happy to help anyone who wants to experiment with that.  If anyone is interested, please chime in on this thread!

The reason I abandoned adding a bite sensor is that all the designs I considered looked a bit difficult to build.  For instance, a while back I had stumbled upon [this video](https://www.youtube.com/watch?v=ag_p8gayrTs) that looked like an interesting design, but that requires many manual tweaks that are very hard to manufacture at scale.  That would probably require a custom mouthpiece and modifications to the neck to route the sensor signals to the RPi.

Probably the only easy part would be the signal processing required for interfacing with the RPi.  If the sensor produces a clean analog output, it can be fed into the RPi via an analog-to-digital converter (ADC) that supports the I2C protocol.  This is exactly how the pressure signal is read, with an MCP3221, an I2C ADC.

Cheers,
